solidity/contracts/token/HypERC20Collateral.sol

```diff
@@ -52,8 +52,9 @@ contract HypERC20Collateral is FungibleTokenRouter, MovableCollateralRouter {
         address _hook,
         address _interchainSecurityModule,
         address _owner
-    ) public virtual initializer {
+    ) public virtual reinitializer(2) {
```

Comment:
This will allow upgrading previous versions of the contract + deploy new versions of the contract because new versions will be deployed as init version 2 directly. However, upgraded contracts will call the _MailboxClient_initialize which I don't think is the expected behavior

Reply:
I like this approach for being more explicit, but I think having multiple initialization functions sets us up for off-chain issues (e.g. forgetting to call `initialize` and only calling `initializeV2` on new routers).